Exclude legacy extensions in the AMO whitelist

crankycoder opened the pull request:
The TAAR addon whitelist was including legacy addons by mistake.  This filters the list further ensuring that an `is_webextension` flag exists and is true for each addon in the whitelist.

Dexterp37 commented on mozetl/taar/taar_amowhitelist.py:
```diff
@@ -64,12 +64,24 @@ def transform(self, json_data):
 
         new_data = {}
         for guid in json_data.keys():
```
You can probably change this to `for guid, addon_data in json_data.items():` to be more succinct.

Dexterp37 commented on mozetl/taar/taar_amowhitelist.py:
```diff
@@ -64,12 +64,24 @@ def transform(self, json_data):
+            if len(current_version_files) == 0:
+                # Only allow webextensions
+                continue
+            else:
```
Is this `else` really needed? If we stumble upon the previous case, then we `continue`. I think we can just drop the `else` to reduce indentation as per our [style guide](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style) (see "General practices").

Dexterp37 commented on mozetl/taar/taar_amowhitelist.py:
```diff
@@ -64,12 +64,24 @@ def transform(self, json_data):
+                    # Only allow webextensions
+                    continue
+
+            rating = addon_data['ratings']['average']
```
Let's be consistent with the code above and use `.get()` whenever possible, eventually catching errors and failing gracefully.

crankycoder commented:
@fbertsch I've added a short comment regarding the JSON we're loading just above the fixture data.

Other than that - I believe I've addressed all changes requested by @Dexterp37
